Src/PCompiler/CompilerCore/Backend/Java/JavaCompiler.cs

```diff
@@ -29,15 +29,16 @@ public IEnumerable<CompiledFile> GenerateCode(ICompilationJob job, Scope scope)
         {
             GenerateBuildScript(job);
 
-            List<ICodeGenerator> generators = new List<ICodeGenerator>()
+            List<JavaSourceGenerator> generators = new List<JavaSourceGenerator>()
             {
-                new MachineGenerator(Constants.MachineDefnFileName),
-                new EventGenerator(Constants.EventDefnFileName),
                 new TypesGenerator(Constants.TypesDefnFileName),
+                new EventGenerator(Constants.EventDefnFileName),
+                new MachineGenerator(Constants.MachineDefnFileName),
                 new FFIStubGenerator(Constants.FFIStubFileName)
             };
 
-            return generators.SelectMany(g => g.GenerateCode(job, scope));
+            CompilationContext ctx = new CompilationContext(job);
+            return generators.SelectMany(g => g.GenerateCode(ctx, scope));
```

Comment:
I meant replacing, job with context?

Reply:
Nope, we're good here in terms of not breaking other stuff - one of the changes was that the "sub-generators" no longer inherit from the global interface, only the top-level one.  (There wasn't really ever a need for that anyway, since they're not visible to the outside world).  As a result, we can change the internal interface to our heart's content.